src/bundle.rs

```diff
@@ -453,18 +543,24 @@ impl Authorized {
         &self.proof
     }
 
-    /// Return the binding signature.
-    pub fn binding_signature(&self) -> &redpallas::Signature<Binding> {
+    /// Return the versioned binding signature.
+    pub fn binding_signature(&self) -> &VerBindingSig {
         &self.binding_signature
     }
 }
 
-impl<V> Bundle<Authorized, V> {
+impl<V, P: OrchardPrimitives> Bundle<Authorized, V, P> {
     /// Computes a commitment to the authorizing data within for this bundle.
     ///
     /// This together with `Bundle::commitment` bind the entire bundle.
-    pub fn authorizing_commitment(&self) -> BundleAuthorizingCommitment {
-        BundleAuthorizingCommitment(hash_bundle_auth_data(self))
+    /// The `sighash_version_map` provides the mapping from each
+    /// `OrchardSighashVersion` to the corresponding `SighashInfo`
+    /// encoding.
+    pub fn authorizing_commitment(
+        &self,
+        sighash_version_map: &BTreeMap<OrchardSighashVersion, Vec<u8>>,
+    ) -> BundleAuthorizingCommitment {
+        BundleAuthorizingCommitment(hash_bundle_auth_data(self, sighash_version_map))
```

Comment:
This feels wrong to me; it implies that the `sighashInfo` comes from outside the transaction, but ZIP 230 is clear that it is encoded as part of `OrchardSignature` and thus should already be present within `self`.

Per my earlier review comment, after refactoring `OrchardSighashVersion` into `OrchardSighashKind ` as I suggested, `sighashInfo` *would* be stored within `self` in some in-memory representation, and then I think instead of passing a map here, we'd want to pass a closure that takes the `OrchardSighashKind` and returns the encoding of its `sighashInfo`.